CONTRIBUTING.md

```diff
@@ -1,8 +1,38 @@
+## Contributing
+
 To find our what this project is about, read the [README.md](README.md)
 
 Via issues / pull requests - https://github.com/Zeckie/ProxyAuth/
 
 Significant changes / new features should be [discussed first](https://github.com/Zeckie/ProxyAuth/discussions) before
 implementing.
 
-The main language of the project is Java (NOT Kotlin!).
\ No newline at end of file
+## Building
+
+### With Gradle
+
+ProxyAuth can be built using Gradle, and includes the
+[Gradle Wrapper](https://docs.gradle.org/7.2/userguide/gradle_wrapper.html). Note that Gradle has different
+requirements, such as not currently supporting Java 17 or later.
+
+- On Windows:
+  `gradlew build`
+- On Linux / Unix:
+  `./gradlew build`
+
+Gradle will create the jar file in `build/libs` (e.g. `build/libs/ProxyAuth-0.1.0.jar`)
+
+### Without Gradle
+
+It does not rely on any third-party libraries, so can be built using any Java IDE, or even javac.
+
+`C:\example\ProxyAuth\src\main\java>javac -d ../bin proxyauth/*.java`
+
+## Design
+
+The main language of the project is Java (NOT Kotlin!).
```

Comment:
I agree. We probably should go further and mandate Java for all code that can be written in Java.

Reply:
We could ditch Gradle and include a jar that builds the project instead.

Reply:
https://stackoverflow.com/questions/12766808/why-is-javac-source-code-is-written-in-java